src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.CompilationTranslationAction.cs

```diff
@@ -12,7 +14,20 @@ internal partial class SolutionState
     {
         private abstract partial class CompilationTranslationAction
         {
-            public abstract Task<Compilation> InvokeAsync(Compilation oldCompilation, CancellationToken cancellationToken);
+            public virtual Task<Compilation> TransformCompilationAsync(Compilation oldCompilation, CancellationToken cancellationToken)
+            {
+                return Task.FromResult(oldCompilation);
```

Comment:
seems like a bug waiting to happen :)   but maybe it's obvious from subclasses they need to overide thsi?

Reply:
That was my guess -- things will be pretty broken otherwise. :-)